### PR TITLE
Add missing factory for line items

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A
| --------------- | -----
| Branch?         | 1.5
| Bug fix?        | kinda
| New feature?    | kinda
| Related tickets | related #416

### CODE REVIEW

**Review by oallain — APPROVED** [review]

**Review by Wojdylak — APPROVED** [review]

File: spec/Converter/LineItem/OrderItemUnitLineItemsConverterSpec.php [hunk 1]
```diff
@@ -65,25 +106,91 @@ function it_converts_unit_refunds_to_line_items(
...
+        $firstLineItem
+            ->compare($secondLineItem)
+            ->willReturn(false);
+        $firstLineItem
+            ->compare($thirdLineItem)
+            ->willReturn(false);
+
+        $secondLineItem
+            ->compare($thirdLineItem)
+            ->willReturn(true);
```
Comment (GSadee) [inline spec/Converter/LineItem/OrderItemUnitLineItemsConverterSpec.php]:
```suggestion
        $firstLineItem->compare($secondLineItem)->willReturn(false);
        $firstLineItem->compare($thirdLineItem)->willReturn(false);
        $secondLineItem->compare($thirdLineItem)->willReturn(true);
```

File: spec/Converter/LineItem/ShipmentLineItemsConverterSpec.php [hunk 2]
```diff
@@ -38,12 +43,55 @@ function it_implements_line_items_converter_interface(): void
...
+        $lineItemFactory->createWithData(
+            'Galaxy post',
+            1,
+            500,
+            575,
+            500,
+            575,
+            75,
+            '15%',
+        )->willReturn($lineItem);
```
Comment (GSadee) [inline spec/Converter/LineItem/ShipmentLineItemsConverterSpec.php]:
```suggestion
        $lineItemFactory
            ->createWithData('Galaxy post', 1, 500, 575, 500, 575, 75, '15%')
            ->willReturn($lineItem)
        ;
```

File: spec/Converter/LineItem/OrderItemUnitLineItemsConverterSpec.php [hunk 3]
```diff
@@ -141,11 +258,10 @@ function it_throws_an_exception_if_there_is_no_shipping_adjustment_with_given_id
 
         $this
             ->shouldThrow(\InvalidArgumentException::class)
-            ->during('convert', [[$unitRefund]])
-        ;
+            ->during('convert', [[$unitRefund]]);
```
Comment (GSadee) [inline spec/Converter/LineItem/OrderItemUnitLineItemsConverterSpec.php]: Not needed change

File: src/Converter/LineItem/OrderItemUnitLineItemsConverter.php [hunk 4]
```diff
@@ -27,7 +28,16 @@ final class OrderItemUnitLineItemsConverter implements LineItemsConverterUnitRef
     public function __construct(
         private RepositoryInterface $orderItemUnitRepository,
         private TaxRateProviderInterface $taxRateProvider,
+        private ?LineItemFactoryInterface $lineItemFactory,
```
Comment (GSadee) [inline src/Converter/LineItem/OrderItemUnitLineItemsConverter.php]:
```suggestion
        private ?LineItemFactoryInterface $lineItemFactory = null,
```

File: src/Converter/LineItem/ShipmentLineItemsConverter.php [hunk 5]
```diff
@@ -29,7 +30,16 @@ final class ShipmentLineItemsConverter implements LineItemsConverterUnitRefundAw
     public function __construct(
         private RepositoryInterface $adjustmentRepository,
         private TaxRateProviderInterface $taxRateProvider,
+        private ?LineItemFactoryInterface $lineItemFactory,
```
Comment (GSadee) [inline src/Converter/LineItem/ShipmentLineItemsConverter.php]:
```suggestion
        private ?LineItemFactoryInterface $lineItemFactory = null,
```

File: spec/Converter/LineItem/OrderItemUnitLineItemsConverterSpec.php [hunk 6]
```diff
@@ -53,6 +60,40 @@ function it_converts_unit_refunds_to_line_items(
...
+        $this->convert([$unitRefund])->shouldBeLike([$lineItem]);
+    }
+
+    function it_converts_unit_refunds_to_line_items_without_using_factory(
+        RepositoryInterface $orderItemUnitRepository,
+        TaxRateProviderInterface $taxRateProvider,
+        OrderItemUnitInterface $orderItemUnit,
+        OrderItemInterface $orderItem,
+    ): void {
+        $this->beConstructedWith($orderItemUnitRepository, $taxRateProvider, null);
```
Comment (GSadee) [inline spec/Converter/LineItem/OrderItemUnitLineItemsConverterSpec.php]:
```suggestion
        $this->beConstructedWith($orderItemUnitRepository, $taxRateProvider);
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Jan! :1st_place_medal:

**Pull request merged**